### ISSUE TITLE
Can't download lantern from https://torch-cdn.mlverse.org when install torch 0.13.0 in win11

### ISSUE THREAD

**Issue by SpatLyu** [mlverse/torch#1172]
![image](https://github.com/mlverse/torch/assets/118346681/d7c13e41-724f-4f0c-b482-2dea9aba34bf)

I have tried to install the binary directly from the browser and it doesn't work either,Thanks for all possible help,

**Comment by Backlonhw2468** [comment]
I ran into a similar situation. Libtorch downloaded successfully while lantern download failed. Please help.

**Comment by dfalbel** [comment]
Sorry for the disruption. Should be back working now

**Comment by SpatLyu** [comment]
@dfalbel Yes, it can be installed normally now,thanks!

**Comment by danielrodonnell** [comment]
I'm having the same issue with 0.12.0 and 0.14.1.

The problem seems to be that the lantern download link is broken. Note that this is not a firewall/proxy/ssl issue. I've explored that fully.

**Comment by dfalbel** [comment]
@danielrodonnell can you post the error mensagem? Knowing the URL that is failing to download will be useful!

Thanks!

**Comment by danielrodonnell** [comment]
@dfalbel Sure thing. 

> torch::install_torch(cuda_version="12.4")
trying URL 'https://torch-cdn.mlverse.org/binaries/lantern-0.12.0+cu124-win64.zip'
Error in `download_file()`:
✖ Unable to download from <https://torch-cdn.mlverse.org/binaries/lantern-0.12.0+cu124-win64.zip>
ℹ Please verify that the URL is not blocked by your firewall. See also <https://torch.mlverse.org/docs/articles/installation.html#file-based-download>
Caused by error in `utils::download.file()`:
! cannot open URL 'https://torch-cdn.mlverse.org/binaries/lantern-0.12.0+cu124-win64.zip'

I got the same error when I tried it with  torch v0.14.1.

Again, the error says to make sure it's not blocked by the firewall, but I'm pretty certain that's not the issue. I was able to manually download libtorch without a problem and have successfully bypassed all of the roadblocks my organization's proxy server throws up. Trying to manually download lantern gets me this:

This XML file does not appear to have any style information associated with it. The document tree is shown below.
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied</Message>
</Error>

**Comment by dfalbel** [comment]
The error actually make sense, as back in v0.12.0 we didn't support cuda124. We do support it with torch v0.14.1 though. But the URL should be different from what you see though.

```
https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip
```

Do you have an env var such as `LANTERN_BASE_URL` that is set?

**Comment by danielrodonnell** [comment]
@dfalbel Thanks. I got the same error when I tried to install the latest torch as well. I did try Sys.setenv(LANTERN_BASE_URL="https://torch-cdn.mlverse.org/binaries/"), but that didn't work either.

The link in your last comment takes me to the same "Access Denied" page when I paste it in the browser and the same 403 error when I try from install_torch().

**Comment by danielrodonnell** [comment]
@dfalbel Here's the full error message. I got this after reinstalling the latest torch version using install_github("mlverse/torch")

> torch::install_torch(cuda_version="12.4")
ℹ Additional software needs to be downloaded and installed for torch to work correctly.
trying URL 'https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip'
trying URL 'https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip'
Error in `download_file()`:
✖ Unable to download from <https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip>
ℹ Please verify that the URL is not blocked by your firewall. See also <https://torch.mlverse.org/docs/articles/installation.html#file-based-download>
Caused by error in `utils::download.file()`:
! cannot open URL 'https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip'
Run `rlang::last_trace()` to see where the error occurred.
Warning messages:
1: In utils::download.file(url = url, destfile = destfile) :
  downloaded length 0 != reported length 111
2: In utils::download.file(url = url, destfile = destfile) :
  cannot open URL 'https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0%2Bcu124-win64.zip': HTTP status was '403 Forbidden'
3: In file.remove(tmp) :
  cannot remove file 'C:\Users\DODONN~1\AppData\Local\Temp\RtmpwjowJU\file688bc24e25b76.zip', reason 'No such file or directory'
4: ℹ Failed to install torch, manually run `install_torch()`
Unable to download from <https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip>
Caused by error in `download_file()`:
✖ Unable to download from <https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip>
ℹ Please verify that the URL is not blocked by your firewall. See also <https://torch.mlverse.org/docs/articles/installation.html#file-based-download>
Caused by error in `utils::download.file()`:
! cannot open URL 'https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0+cu124-win64.zip' 
5: In utils::download.file(url = url, destfile = destfile) :
  downloaded length 0 != reported length 111
6: In utils::download.file(url = url, destfile = destfile) :
  cannot open URL 'https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.12.0%2Bcu124-win64.zip': HTTP status was '403 Forbidden'
Warning message:
In file.remove(tmp) :
  cannot remove file 'C:\Users\DODONN~1\AppData\Local\Temp\RtmpwjowJU\file688bc4f82516e.zip', reason 'No such file or directory'

**Comment by dfalbel** [comment]
Sorry, I mistyped the URL, it should be:

```
https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.14.1+cu124-win64.zip
```

which does work.

Are you manually setting this URL?

**Comment by danielrodonnell** [comment]
@dfalbel That worked! Thank you! It may be a good idea to make sure that URL is what the package is trying to download with install_torch(). I'm not sure it's doing that right. Thank you x1000 for all your help :)

**Comment by danielrodonnell** [comment]
@dfalbel Maybe this is for another issue, but...

1) The URL you gave me works and I successfully downloaded the .zip.
2) install_torch(cuda_version="12.4") now returns this error:

> torch::install_torch(cuda_version="12.4")
Warning message:
ℹ torch failed to start, restart your R session to try again.
ℹ You might need to reinstall torch using `install_torch()`
✖ C:\Users\dodonnell\AppData\Local\R\win-library\4.4\torch/lib\lantern.dll - The specified procedure could not be found.
Caused by error in `cpp_lantern_init()`:
  C:\Users\dodonnell\AppData\Local\R\win-library\4.4\torch/lib\lantern.dll - The specified procedure could not be found.

**Comment by danielrodonnell** [comment]
@dfalbel 

> sessionInfo()
R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] torch_0.14.1

loaded via a namespace (and not attached):
 [1] processx_3.8.5    bit_4.5.0.1       compiler_4.4.1    magrittr_2.0.3    cli_3.6.3         tools_4.4.1       rstudioapi_0.16.0 Rcpp_1.0.14      
 [9] bit64_4.6.0-1     coro_1.1.0        callr_3.7.6       ps_1.8.1          rlang_1.1.5

**Comment by dfalbel** [comment]
I think that could happen if you don't have the necessary CUDA libraries findable. Can you try installing the CPU version instead, just so we can isolate the problems?

You should be able to use a URL like:

```
https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.14.1+cpu-win64.zip
```

**Comment by danielrodonnell** [comment]
@dfalbel 

Here's what I've done:

>Sys.setenv(TORCH_URL="https://download.pytorch.org/libtorch/cpu/libtorch-win-shared-with-deps-2.6.0%2Bcpu.zip")
>Sys.setenv(LANTERN_URL="https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.14.1+cpu-win64.zip")

>remotes::install_github("mlverse/torch")

> torch::install_torch()
Warning message:
ℹ torch failed to start, restart your R session to try again.
ℹ You might need to reinstall torch using `install_torch()`
✖ C:\Users\dodonnell\AppData\Local\R\win-library\4.4\torch/lib\lantern.dll - The specified procedure could not be found.
Caused by error in `cpp_lantern_init()`:
  C:\Users\dodonnell\AppData\Local\R\win-library\4.4\torch/lib\lantern.dll - The specified procedure could not be found.

**Comment by dfalbel** [comment]
hmm. You can't use libtorch v2.6.0, you should use 2.5.1. Maybe that's the issue?

**Comment by danielrodonnell** [comment]
@dfalbel You are correct! Here's what worked:

>Sys.setenv(TORCH_URL="https://download.pytorch.org/libtorch/cu124/libtorch-win-shared-with-deps-2.5.1%2Bcu124.zip")
>Sys.setenv(LANTERN_URL="https://torch-cdn.mlverse.org/binaries/refs/heads/cran/v0.14.1/latest/lantern-0.14.1+cu124-win64.zip")

>remotes::install_github("mlverse/torch")

>torch::install_torch(cuda_version="12.4")

> cuda_is_available()
[1] TRUE
